packages/ui/src/components/va-modal/VaModal.vue

```diff
@@ -242,12 +243,15 @@ export default defineComponent({
       // Supposedly solves some case when background wasn't shown.
       // As a side effect removes background from nested modals.
 
-      if (!props.overlay || !isLowestLevelModal.value) { return }
+      if (!props.overlay) { return }
 
-      return {
-        'background-color': `rgba(0, 0, 0, ${props.overlayOpacity})`,
-        'z-index': props.zIndex && Number(props.zIndex) - 1,
-      } as StyleValue
+      if (isTopLevelModal.value || props.isShowNestedOverlay) {
+        return {
+          'background-color': `rgba(0, 0, 0, ${props.overlayOpacity})`,
```

Comment:
Would be nice if we can make nestedOverlay opacity smaller... Because with 3 nested modals overlay become so much darken. 


 How about using css variables here instead of hardcoding?
 
 ```
 :root {
    --va-modal-overlay-color: rgb(0, 0, 0);
    --va-modal-overlay-opacity: 0.6;
    --va-modal-overlay-nested-opacity: 0.1;
 }
 ```
 
 We can remove backgroundOpacity prop.

Reply:
By removing `overlayOpacity` from props, one cannot control the overlay that will be okay ?

Reply:
I think user still have `overlay` option to turn it off. In case user wants opacity to be different from 0.6, it can be changed using CSS variables.

Reply:
```
 if (isTopLevelModal.value || props.ShowNestedOverlay) {
        return {
          'background-color': 'var(--va-modal-overlay-color)',
          opacity: `${props.ShowNestedOverlay ? 'var(--va-modal-overlay-nested-opacity)' : 'var(--va-modal-overlay-opacity)'} `,
          'z-index': props.zIndex && Number(props.zIndex) - 1,
        } as StyleValue
      }
```

Something like this ?

Reply:
https://github.com/epicmaxco/vuestic-ui/assets/26655564/329cd724-7669-41b0-8a77-ec27a706fd31

By using this approach I'm able to see the black background flickering on Clicking Cancel of Nested Dialog

Reply:
> Screen.Recording.2023-09-05.at.9.39.24.PM.mov 
> By using this approach I'm able to see the black background flickering on Clicking Cancel of Nested Dialog

If you want me to help you, please update your PR.

Reply:
have updated my PR